### ISSUE TITLE
Linting of the step definition files

### ISSUE THREAD

**Issue by nene** [vsiakka/gherkin-lint#69]
What are your thoughts about pulling JavaScript files also into the linting process?

Two things I'd like to check with a linter:

- Unused step definitions (a step is defined in JS, but not used in any *.feature file).
- Step definition keyword doesn't match with keyword used for that step in a *.feature file (e.g. a `Then` step is defined, but it's used as a `When` step).

I guess this question comes down to whether gherkin-lint should be agnostic of the implementation language. Would we expect gherkin-lint to be used outside the JavaScript ecosystem?

The other part of the question is whether these step definition files are consistent enough even inside JavaScript world.

[cucumber-js docs](https://github.com/cucumber/cucumber-js/blob/master/docs/support_files/step_definitions.md) show the following step definitions:

```js
var {defineSupportCode} = require('cucumber');

defineSupportCode(function({Then, When}) {
  Then(/^Then the response status is (.*)$/, function (status) {
    // ...
  });
});
```

While [chimp docs](https://chimp.readme.io/v1.0/docs/tutorial) define steps as follows:

```js
module.exports = function() {
  this.Then(/^Then the response status is (.*)$/, function (status) {
    // ...
  });
}
```

Cucumber.js docs actually also show that `this.Then()` syntax can also be used. Perhaps the vice versa is true for Chimp. I understand Chimp actually uses the same Cucumber.js under the hood.

**Comment by xgbuils** [comment]
What about creating a cucumber-js-lint project and keep this project agnostic? I think it could be good separating responsabilities.

Cheers!

**Comment by josemenveg** [comment]
I'm not agree.
Nobody says that the the steps implementations must be in Javascript, I've seen projects in Ruby and Java.

As @xgbuils said, it ould be a different project
